lib/src/signed_video_h26x_sign.c

```diff
@@ -240,12 +239,9 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
       // Fallback to GOP-level signing
       payload_size -= document_size;
       self->gop_info->list_idx = -1;  // Reset hash list size to exclude it from TLV
-      optional_document_size = tlv_list_encode_or_get_size(
-          self, optional_document_encoders, num_optional_document_encoders, NULL);
-      mandatory_document_size = !self->is_golden_sei
-          ? tlv_list_encode_or_get_size(
-                self, mandatory_document_encoders, num_man_doc_encoders, NULL)
-          : 0;
+      mandatory_document_size =
+          tlv_list_encode_or_get_size(self, mandatory_tags, num_mandatory_tags, NULL);
+      if (self->is_golden_sei) mandatory_document_size = 0;
```

Comment:
I think it is better to check if `mandatory_document_size > 0` in the if on line 238, otherwise you'll simply enter and recalculate something that in the end again is set to zero.